pyro/infer/autoguide/guides.py

```diff
@@ -347,6 +347,145 @@ def median(self, *args, **kwargs):
         return self(*args, **kwargs)
 
 
+class AutoNormal(AutoGuide):
+    """This is AutoNormal, but it uses the same shape constraint logic found
+    in AutoMultivariateNormal and AutoDiagonalNormal.
+    """
+    def __init__(self, model, init_loc_fn=init_to_median, prefix='auto'):
+        #  Does this accept a 'prefix' arg?  It doesn't look like it.
+        self.init_loc_fn = init_loc_fn
+        self.prefix = prefix
+        model = InitMessenger(self.init_loc_fn)(model)
+        super().__init__(model)
+
+    def _setup_prototype(self, *args, **kwargs):
+        super()._setup_prototype(*args, **kwargs)
+
+        self._unconstrained_shapes = {}
+        self._cond_indep_stacks = {}
+
+        # Initialize guide params
+        for name, site in self.prototype_trace.iter_stochastic_nodes():
+            # Collect the shapes of unconstrained values.
+            # These may differ from the shapes of constrained values.
+            constrained_shape = site["value"].shape
+            unconstrained_shape = biject_to(site["fn"].support).inv(site["value"]).shape
+
+            event_dim = site["fn"].event_dim + len(unconstrained_shape) - len(constrained_shape)
+
+            site_batch_shape = site["fn"].batch_shape
+
+            self._unconstrained_shapes[name] = unconstrained_shape
+            unconstrained_shape_for_params = broadcast_shape(
+                unconstrained_shape,
+                site_batch_shape + (1,) * event_dim
+            )
+
+            # Collect independence contexts.
+            self._cond_indep_stacks[name] = site["cond_indep_stack"]
+
+            #  Set up params
+            loc_name = "{}_{}_{}".format(self.prefix, name, 'loc')
+            scale_name = "{}_{}_{}".format(self.prefix, name, 'scale')
+
+            pyro.param(
+                loc_name,
+                lambda: site["value"].new_zeros(unconstrained_shape_for_params)
+            )
+            pyro.param(
+                scale_name,
+                lambda: site["value"].new_ones(unconstrained_shape_for_params),
+                constraint=constraints.positive
+            )
```

Comment:
The major change in 1.0 was to switch from `pyro.param` to `PyroModule`s to support serialization and serving. I think you can update this block as follows:
```py
self.locs = PyroModule()
self.scales = PyroModule()
for name, site in self.prototype_trace.iter_stochastic_nodes():
    ...
    if hasattr(self.locs, name):
        continue
    init_loc = site["value"].new_zeros(unconstrained_shape_for_params)
    init_scale = site["value"].new_full(unconstrained_shape_for_params, self.init_scale)
    setattr(self.locs, name, nn.Parameter(init_loc))
    setattr(self.scales, name, PyroParam(init_scale, constraints.positive))
```